.github/workflows/multi-arch-build.yaml

```diff
@@ -0,0 +1,175 @@
+      - name: amd64 container sniff test
+        id: sniff_test
+        run: |
+          VERSION_OUTPUT="$(docker run localhost:5000/podman/${{ matrix.source }} \
+                            podman --storage-driver=vfs version)"
+          echo "$VERSION_OUTPUT"
+          VERSION=$(grep -Em1 '^Version: ' <<<"$VERSION_OUTPUT" | awk '{print $2}')
+          test -n "$VERSION"
+          echo "::set-output name=version::${VERSION}"
+
+      # Generate image FQINs, labels, check whether to push
+      - name: Generate image information
+        id: image_info
+        run: |
+          if [[ "${{ matrix.source }}" == 'stable' ]]; then
+            # quay.io/podman/stable:vX.X.X
+            PUSH=$(skopeo list-tags \
+                     docker://${{ env.PODMAN_QUAY_REGISTRY }}/stable | \
+                   jq -r .Tags[] | \
+                   grep -q "^${VERSION}$" && \
+                   echo 'false' || echo 'true')
+            FQIN='${{ env.PODMAN_QUAY_REGISTRY }}/stable:v${{ steps.sniff_test.outputs.version }}' # workaround vim syntax-hilighting bug: '
+            # Only push if version tag does not exist
+            if [[ "$PUSH" == "true" ]]; then
+              echo "Will push $FQIN"
+              echo "::set-output name=podman_push::${PUSH}"
+              echo "::set-output name=podman_fqin::${FQIN}"
+            fi
+
+            # quay.io/contaners/podman:vX.X.X
+            unset PUSH
+            PUSH=$(skopeo list-tags \
+                       docker://${{env.CONTAINERS_QUAY_REGISTRY}}/podman | \
+                   jq -r .Tags[] | \
+                   grep -q "^${VERSION}$" && \
+                   echo 'false' || echo 'true')
+            FQIN='${{ env.CONTAINERS_QUAY_REGISTRY}}/podman:v${{ steps.sniff_test.outputs.version }}' # workaround vim syntax-hilighting bug: '
+            # Only push if version tag does not exist
+            if [[ "$PUSH" == "true" ]]; then
+              echo "Will push $FQIN"
+              echo "::set-output name=containers_push::${PUSH}"
+              echo "::set-output name=containers_fqin::$FQIN"
+            fi
+          else  # upstream/testing/stable podman image
+            P_FQIN='${{ env.PODMAN_QUAY_REGISTRY }}/${{ matrix.source }}:master' # workaround vim syntax-hilighting bug: '
+            C_FQIN='${{ env.CONTAINERS_QUAY_REGISTRY}}/podman:master' # workaround vim syntax-hilighting bug: '
+            echo "Will push $P_FQIN and $C_FQIN"
+            echo "::set-output name=podman_fqin::${P_FQIN}"
+            echo "::set-output name=containers_fqin::${C_FQIN}"
+            # Always push 'master' tag
+            echo '::set-output name=podman_push::true'
+            echo '::set-output name=containers_push::true'
+          fi
+
+          # Hack to set $LABELS env. var. in _future_ steps.
+          # https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable
+          cat << EOF | tee $GITHUB_ENV
```

Comment:
It should raise eyebrows...(unfortunately) like many other aspects, this construct is idiomatic to github-actions :frowning_face:  That's why I dropped the docs URL comment, and I'm glad that found some use already :smiley: 

The file never exists, the step is suppose to create it to that triggers parsing of the contents.    The `echo '::set-output name=foo::bar'` idiom only works for a simple single-line value (because reasons).  There's no other way to pass a ***multi-line*** value to subsequent actions (i.e. `docker/build-push-action@v2`).  This is the best M$ GitHub could come up with :confounded: 

In buildah, the matching step uses a simple here-document redirect to a file.  As you guessed, that means no record of the value shows in the logs.  This is/was hiding an actual bug.  I only discovered it by manually doing a `skopeo inspect` on one of the images :angry:  So I changed it here to use `tee` and make it show up in the logs to prevent future headaches.

TBH I'm not sure how to make this section any clearer, it's kind of broken by the workflow "language" design :cry:

Reply:
Here's an [example of the log output from this section](https://github.com/containers/podman/runs/2403981082#step:8:58) as-is.

Reply:
I've tried twice to understand that, and failed. But it's not important, I don't really need to understand it. My point is: all the github examples use `>>` (append). In the line above, you use `tee` (clobber), not `tee -a` (append). This is inconsistent with the examples documented in github.com itself, and could potentially overwrite something important, so I wanted to make sure that your use is deliberate.